File: src/main/javabackup/de/thatsich/intellie/common/registries/IRegistry.java
package de.thatsich.intellie.common.registries;

/**
 All registies need to be able to register their stuff in the end

 @author thatsIch
 @since 25.03.2014. */
public interface IRegistry<T>
{
	void add( T elem );

	/**
	 Registers that particular part
	 */
	void register();
}

File: src/main/scala/de/thatsich/minecraft/common/registries/IRegistry.scala
package de.thatsich.minecraft.common.registries

import de.thatsich.minecraft.common.logger.ALog


/**
 *
 *
 * @author thatsIch
 * @since 06.04.2014.
 */
trait IRegistry[ T ]
{
	/**
	Adds an element to the registry.
	It will get registered later on

	  * @param elem New element in the registry
	  */
	def add(elem: T)(implicit log: ALog)

	/**
	Registers that particular part
	  */
	def register()(implicit log: ALog)
}
